File: 프로그래머스/2/42584. 주식가격/주식가격.py
from collections import deque

def solution(prices):
    answer = []
    queue = deque(prices)
    
    while queue :
        current = queue.popleft()
        cnt = 0
        
        tmp = False
        for next in queue :
            cnt += 1
            if current > next :
                answer.append(cnt)
                tmp = True
                break
                
        if not tmp :
            answer.append(cnt)
    
    return answer

File: 프로그래머스/2/42584. 주식가격/주식가격.java
class Solution {
    public int[] solution(int[] prices) {
        int[] answer = new int[prices.length];
        
        for (int i=0; i<prices.length; i++) {
            int tmp = 0;
            int current = prices[i];
            for (int j=i+1; j<prices.length; j++) {
                int next = prices[j];
                tmp++;
                if (current > next) {
                    break;
                }
            }
            answer[i] = tmp;
        }
        
        return answer;
    }
}